### PR TITLE
correctly parse the name from the package.json - fixes #12

### LINKED ISSUE #12: Wrong Node Module name after extraction
if a module name is not the first name attribute in the package.json, the module is extracting with the wrong name. 

E.g. the module [util](https://github.com/defunctzombie/node-util) declares the author's name "Joyent" before the module name and after extraction the module has a name "Joyent" instead of "util".

You can use https://github.com/ilucker/webjars-node-extract to reproduce the issue. 
Run `>sbt web-assets:webJarsNodeModules` and check `/target/web/node-modules/main/webjars`

### PR DESCRIPTION
**Pull request by jamesward**

### CODE REVIEW

**Comment by jamesward** [comment]
Review request: @jroper @mwanji @huntc

File: pom.xml [hunk 1]
```diff
@@ -60,6 +60,11 @@
             <version>1.9</version>
         </dependency>
         <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
```
Comment (jroper) [inline pom.xml]: If you wanted to minimise dependencies, you could just use jackson-core, and pull the name out using the jackson token parser.

File: pom.xml [hunk 2]
```diff
@@ -60,6 +60,11 @@
...
+            <version>2.3.3</version>
```
Comment (jroper) [inline pom.xml]: This is very old... Jackson is now up to 2.7, any reason for not using the latest?

File: src/main/java/org/webjars/WebJarExtractor.java [hunk 3]
```diff
@@ -320,28 +325,14 @@ private String getFileNodeModuleIdEntry(File packageJsonFile) throws IOException
...
-                    }
-                    moduleId = sb.toString();
-                }
-            }
-        }
-        return moduleId;
+    String getJsonNodeModuleId(String packageJson) throws IOException {
+        ObjectMapper mapper = new ObjectMapper()
+                .configure(ALLOW_UNQUOTED_FIELD_NAMES, true)
+                .configure(ALLOW_SINGLE_QUOTES, true);
```
Comment (jroper) [inline src/main/java/org/webjars/WebJarExtractor.java]: Do you have any examples of where these are actually needed?  From [the docs](https://docs.npmjs.com/files/package.json):

> It must be actual JSON, not just a JavaScript object literal.

**Comment by jamesward** [comment]
Thanks @jroper. I've updated the PR based on your feedback.

**Comment by jroper** [comment]
LGTM.

**Pull request merged**